sbpy/utils/dastcom5.py

```diff
@@ -0,0 +1,575 @@
+        ("COMEST", "|S14"),
+        ("IREF", "|S10"),
+        ("COMNAM", "|S29"),
+    ]
+)
+
+SBPY_LOCAL_PATH = os.path.join(os.path.expanduser("~"), ".sbpy")
+DBS_LOCAL_PATH = os.path.join(SBPY_LOCAL_PATH, "dastcom5", "dat")
+AST_DB_PATH = os.path.join(DBS_LOCAL_PATH, "dast5_le.dat")
+COM_DB_PATH = os.path.join(DBS_LOCAL_PATH, "dcom5_le.dat")
+
+FTP_DB_URL = "ftp://ssd.jpl.nasa.gov/pub/ssd/"
+
+
+def asteroid_db():
+    """Return complete DASTCOM5 asteroid database.
+
+    Returns
+    -------
+    database : numpy.ndarray
+        Database with custom dtype.
+
+    """
+    with open(AST_DB_PATH, "rb") as f:
+        f.seek(835, os.SEEK_SET)
+        data = np.fromfile(f, dtype=AST_DTYPE)
+    return data
+
+
+def comet_db():
+    """Return complete DASTCOM5 comet database.
+
+    Returns
+    -------
+    database : numpy.ndarray
+        Database with custom dtype.
+
+    """
+    with open(COM_DB_PATH, "rb") as f:
+        f.seek(976, os.SEEK_SET)
+        data = np.fromfile(f, dtype=COM_DTYPE)
+    return data
+
+
+def orbit_from_name(name):
+    """Return :py:class:`~astropy.table.Table` given a name.
+
+    Retrieve info from JPL DASTCOM5 database.
+
+    Parameters
+    ----------
+    name : str
+        NEO name.
+
+    Returns
+    -------
+    Table : ~astropy.table.Table
+        Near Earth Asteroid/Comet orbit parameters, all stacked.
+
+    """
+    records = record_from_name(name)
+    orbits = [orbit_from_record(rec) for rec in records]
+    tbl = Table(orbits)
+    return tbl
+
+
+def orbit_from_record(record):
+    """Return :py:class:`~astropy.table.Table` given a record.
+
+        Retrieve info from JPL DASTCOM5 database.
+
+        Parameters
+        ----------
+        record : int
+            Object record.
+
+        Returns
+        -------
+        Table : ~astropy.table.Table
+            Near Earth Asteroid/Comet orbit parameters.
+
+    """
+    body_data = read_record(record)
+    a = body_data["A"].item()
+    ecc = body_data["EC"].item()
+    inc = body_data["IN"].item()
+    raan = body_data["OM"].item()
+    argp = body_data["W"].item()
+    m = body_data["MA"].item()
+    epoch = Time(body_data["EPOCH"].item(), format="jd", scale="tdb")
+    column2 = (record, a, ecc, inc, raan, argp, m, epoch)
+    column1 = ("record", "a", "ecc", "inc", "raan", "argp", "m", "EPOCH")
+    data = Table(rows=[column1, column2])
+
+    return data
+
+
+def record_from_name(name):
+    """Search `dastcom.idx` and return logical records that
+    match a given string.
+
+    Body name, SPK-ID, or alternative designations can be used.
+
+    Parameters
+    ----------
+    name : str
+        Body name.
+
+    Returns
+    -------
+    records : list (int)
+        DASTCOM5 database logical records matching str.
+
+    """
+    records = []
+    lines = string_record_from_name(name)
+    for line in lines:
+        records.append(int(line[:6].lstrip()))
+    return records
+
+
+def string_record_from_name(name):
```

Comment:
this function should be mentioned in the documentation

Reply:
Sure